File: backend/domain/src/main/java/ru/itmo/saferoad/domain/entity/Level.java
package ru.itmo.saferoad.domain.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;

import java.util.Objects;

@Entity
@Getter
@Setter
@Table(name = "levels")
public class Level {

	@Id
	@NonNull
	@Column(name = "id", nullable = false)
	private Long id;

	@NonNull
	@Column(name = "number", nullable = false)
	private Integer number;

	@NonNull
	@Column(name = "title", nullable = false)
	private String title;

	@NonNull
	@Column(name = "xp_threshold", nullable = false)
	private Integer xpThreshold;

	@Override
	public boolean equals(Object o) {
		if (this == o) {
			return true;
		}
		if (!(o instanceof Level level)) {
			return false;
		}
		return Objects.equals(getId(), level.getId());
	}

	@Override
	public int hashCode() {
		return Objects.hash(getId());
	}
}

File: safe-road-mobile/lib/data/models/gamification/level.dart
class Level {
  final int number;
  final String title;
  final int xpThreshold;

  Level({
    required this.number,
    required this.title,
    required this.xpThreshold,
  });

  factory Level.fromJson(Map<String, dynamic> json) {
    return Level(
      number: json['number'],
      title: json['name'],
      xpThreshold: json['xpThreshold'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'number': number,
      'title': title,
      'xpThreshold': xpThreshold,
    };
  }
}
